hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/OverallSkewChecker.java

```diff
@@ -0,0 +1,109 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.master.balancer;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.ServerName;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.RegionInfo;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The cluster status checker for {@link StochasticLoadBalancer}, if the skew table counts up to a
+ * configured ratio, the simplified balancer will be executed, it is aimed to balance the regions as
+ * soon as possible. For example, when there are large-scaled restart of RSes, or expansion for
+ * groups or cluster, we hope the balancer can execute as soon as possible, but the
+ * StochasticLoadBalancer may need a lot of time to compute costs.
+ */
+@InterfaceAudience.Private
+class OverallSkewChecker implements StochasticLoadBalancer.ClusterStatusBalanceChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(OverallSkewChecker.class);
+
+  private static final String BALANCER_OVERALL_SKEW_PERCENT_KEY =
+    "hbase.master.balancer.overall.skew.percentage";
+
+  private Configuration conf;
+
+  public OverallSkewChecker(Configuration conf) {
+    this.conf = conf;
+  }
+
+  @Override
+  public boolean
+    needsCoarseBalance(Map<TableName, Map<ServerName, List<RegionInfo>>> loadOfAllTable) {
+    if (this.conf == null) {
+      LOG.warn("Configuration should not be null");
+      return false;
+    }
+    if (isClusterSkew(loadOfAllTable)) {
+      LOG.info("Balancer checker found cluster is skew, will perform force balance, "
+        + "table count is {}", loadOfAllTable.size());
+      return true;
+    }
+    LOG.debug("Balancer checked cluster, it is not skew now, tables count is {}",
+      loadOfAllTable.size());
+    return false;
+  }
+
+  public boolean isClusterSkew(Map<TableName, Map<ServerName, List<RegionInfo>>> loadOfAllTable) {
+    if (loadOfAllTable == null || loadOfAllTable.isEmpty()) {
+      return false;
+    }
+    int skewCount = 0;
+    for (Map.Entry<TableName, Map<ServerName, List<RegionInfo>>> entry : loadOfAllTable
+      .entrySet()) {
+      if (isSkew(entry.getValue())) {
+        LOG.info("Table: " + entry.getKey().getNameAsString() + " regions count is skew");
```

Comment:
Basing this solely on region counts may be a tricky change to roll out ubiquitously, because balancer cost configurations support heavy emphasis on other factors. For example, at my day job we primarily balance based on storefile size, and don't care much about region counts from the balancer's perspective